src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.cpp

```diff
@@ -63,21 +64,42 @@ CHIP_ERROR Instance::Init()
         ChipLogError(Zcl, "Microwave Oven Control: The cluster with ID %lu was not enabled in zap.", long(mClusterId));
         return CHIP_ERROR_INVALID_ARGUMENT;
     }
-
+    // Check if the feature bit is mutually exclusive
+    if (mFeature != to_underlying(MicrowaveOvenControl::Feature::kPowerAsANumber) &&
+        mFeature != to_underlying(MicrowaveOvenControl::Feature::kPowerInWatts))
+    {
+        ChipLogError(Zcl, "Microwave Oven Control: feature bits error, one and only one of the feature bits must be true");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
     ReturnErrorOnFailure(InteractionModelEngine::GetInstance()->RegisterCommandHandler(this));
     VerifyOrReturnError(registerAttributeAccessOverride(this), CHIP_ERROR_INCORRECT_STATE);
 
+    if (HasFeature(to_underlying(MicrowaveOvenControl::Feature::kPowerInWatts)))
+    {
+        mSupportedPowerLevels = GetSupportedPowerLevels();
+    }
     return CHIP_NO_ERROR;
 }
 
-uint32_t Instance::GetCookTime() const
+bool Instance::HasFeature(uint32_t feature) const
```

Comment:
Again, why are we using weakly typed things for features?  There's a reason we have a strongly typed enum for it.